Update random-access-storage to 2.0.0

jackjennings opened the pull request:
~*Submitting this pull request because I need some help resolving an issue*~ Update: resolved.

## Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->
- [x] tests pass
- [x] tests and/or benchmarks are included
- [x] documentation is changed or added

## Context
<!-- Is this related to any GitHub issue(s)? -->

This pull request is in regards to work on this issue to update `usize` to `u64`: https://github.com/datrs/random-access-storage/issues/6

As this package would need to update to the upcoming `3.0.0` of `random-access-storage`, I'm attempting to first update to `2.0.0`, which requires adding a handful of additional methods to the `RandomAccess` trait.

I've added `len` a `is_empty`, but `truncate` is troublesome. I assumed that I could make this implementation `del(0, length)`, but this throws an error when the `offset` provided to `del` is zero. I'm unsure if this is a bug in `del`, or if I need to provide a more complex implementation of `truncate`.

```
thread 'can_truncate_lt' panicked at 'attempt to subtract with overflow', src/lib.rs:180:19
```

I've added failing tests for `truncate` based on the tests in [`random-access-disk`](https://github.com/datrs/random-access-disk/blob/f9b34735e79020ff2f39bef510026f5cf8bf02b3/tests/test.rs#L52-L119).

`dependabot` attempted to make this change previously in this pull request: https://github.com/datrs/random-access-memory/pull/9

## Semver Changes
<!-- Which semantic version change would you recommend? -->

This would likely be a minor version change, as additional functionality is added, while existing functionality is unchanged.

jackjennings commented:
Also I was asked to choose what kind of pull request this was from the template. I assume this is supposed to be a label I add, but labels seem to be disabled for non-contributors.

jackjennings commented on tests/test.rs:
```diff
@@ -30,3 +30,94 @@ fn can_read() {
+  // file.truncate(0).unwrap();
+  // assert_eq!(file.is_empty().unwrap(), true);
+  // file.write(0, b"what").unwrap();
+  // assert_eq!(file.is_empty().unwrap(), false);
```
These tests are disabled, as they require `truncate` to work.

yoshuawuyts commented:
@jackjennings these changes look reasonable to me; agree we should probably figure out why truncate doesn't work, as your reasoning seems sound.

If you could make sure tests pass (cargo fmt is failing) then I think we should be good to merge!

jackjennings commented:
@yoshuawuyts I rebased — changed the implementation of `truncate` to `unimplemented!` and removed the tests that were commented that depended on `truncate. I'll opens a separate pull request for the implementation of `del`. Looking at the way the JavaScript version works give me some clues as to what the issue may be.

yoshuawuyts commented:
`v1.0.0` :tada:
